library/core/src/ptr/non_null.rs

```diff
@@ -1583,7 +1585,7 @@ impl<T> NonNull<[T]> {
     #[rustc_const_unstable(feature = "const_ptr_as_ref", issue = "91822")]
     #[requires(self.as_ptr().cast::<T>().align_offset(core::mem::align_of::<T>()) == 0)] // Ensure the pointer is properly aligned
     #[requires(self.len().checked_mul(core::mem::size_of::<T>()).is_some() && self.len() * core::mem::size_of::<T>() <= isize::MAX as usize)] // Ensure the slice size does not exceed isize::MAX
-    // TODO: add a require to check the slice belong to same allocated object with `same_allocation`
```

Comment:
Why are you deleting the TODO comment? If you're planning to address it soon in a follow up PR, that's fine, but if not, it'd be good to keep around for anyone that picks up this work.
(Same comment about the `as_uninit_slice_mut` function).

Reply:
> Why are you deleting the TODO comment? If you're planning to address it soon in a follow up PR, that's fine, but if not, it'd be good to keep around for anyone that picks up this work. (Same comment about the `as_uninit_slice_mut` function).

Sorry I miss this TODO when updating the code, already added in the latest commit. Thanks for the reminding!